helix-view/src/document.rs

```diff
@@ -652,7 +653,8 @@ impl Document {
             path: None,
             encoding,
             has_bom,
-            text,
+            text: text.clone(),
+            original_text: text,
```

Comment:
I think I get your easlier comment now. I didn't mean adding an extra field here. 

I was only talking about situations where `text` is known to be equal to the on-disk version these situations would be:

* when the file is first opened
*  after `:reload`
* after `:write`

in theses cases you should just send `doc.text().clone()` to the diff worker as a diffbase.

But when the diff provider gets changed (lets say by a seggints change) its indeed fine to read the file from disk again.